0.Preprocess_Data/scripts/1.preprocessing_morphology_data.py

```diff
@@ -239,40 +239,75 @@
 
 feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
     "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
-].str.replace("media_ctr_0.0_0_Media_ctr_0_0", "media_ctr_0.0_0_Media_ctr_0.0_0")
+].str.replace("media_ctr_0.0_0_Media_ctr_0_0", "Media", regex=False)
 
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace("media_ctr_0.0_0_Media_0_0", "Media", regex=False)
 
-# In[18]:
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace("media_ctr_0.0_0_Media_ctr_0.0_0", "Media", regex=False)
 
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace(
+    "Flagellin_1.000_0_Disulfiram_1.000_uM",
+    "Flagellin_1.000_ug_per_ml_Disulfiram_1.000_uM",
+    regex=False,
+)
 
-# need to convert to strings to save as parquet
-# if the column is an object then convert it to a string
-for column in feature_df.columns:
-    if feature_df[column].dtype == "object":
-        feature_df[column] = feature_df[column].astype(str)
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace(
+    "Flagellin_1.000_0_DMSO_0.025_%",
+    "Flagellin_1.000_ug_per_ml_DMSO_0.025_%",
+    regex=False,
+)
 
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace(
+    "Flagellin_0.100_ug_per_ml_DMSO_0.000_%",
+    "Flagellin_0.100_ug_per_ml_DMSO_0.025_%",
```

Comment:
A bit confused here, it looks to me here that you are replacing the treatment with 0% DMSO to 0.025% DMSO. Is this correct?

Reply:
Yes, correct there was an error in the metadata